src/Hosting.Services.Web/Middlewares/UserIdentity/IdentityProviders/EmailBasedUserIdentityProvider.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares
+{
+	internal class EmailBasedUserIdentityProvider : IUserIdentityProvider
+	{
+        private static int MaxBytesPerChar { get; } = 4; // UTF-8
+		public int MaxBytesInIdentity { get; } = 256 * MaxBytesPerChar; // maximum email address length plus max bytes per char
+
+        public class UserEmail
+        {
+            public string Email { get; set; } = string.Empty;
+        }
+
+		public Task<(bool success, int bytesWritten)> TryWriteBytesAsync(HttpContext context, Memory<byte> memory)
+		{
+			int bytesWritten = -1;
+            try{
+                context.Request.Body.Seek(0, SeekOrigin.Begin);
+                TextReader reader = new StreamReader(context.Request.Body);
+                UserEmail? email = JsonSerializer.Deserialize<UserEmail>(reader.ReadToEnd());
+                reader.Close();
+                bool success = email != null;
+                if (email != null)
+                {
+                    bytesWritten = 0;
+                    foreach (char c in email.Email)
+                        {
+                            success = success && BitConverter.TryWriteBytes(memory.Span.Slice(bytesWritten, MaxBytesPerChar), c);
+                            bytesWritten += MaxBytesPerChar;
+                        }
+                }
+                return Task.FromResult((success, bytesWritten));
+            }
+            catch {
+                return Task.FromResult((false, -1));
```

Comment:
Can we log the exception here, so that we can investigate if we see the failures in production?

Reply:
Was considering this, but in `IpBasedIdentityProvider`, failures in writing to span are not logged, so didn't log it here. Willing to do so if needed.

Reply:
Resolving this as we cannot pass a logger in while keeping the interface the same as the `TryWriteBytesAsync()` in `IpBasedIdentityProvider`.

Reply:
I believe you should be able to pass in the logger if you create a constructor. If this is created through dependency injection, the object should be automatically populated